src/proto/uprotocol/uresource.rs

```diff
@@ -0,0 +1,57 @@
+/********************************************************************************
+ * Copyright (c) 2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use crate::uprotocol::UResource;
+
+impl UResource {
+    pub fn has_id(resource: &UResource) -> bool {
+        resource.id.is_some()
+    }
+
+    pub fn get_message(resource: &UResource) -> Option<&String> {
+        match &resource.message {
```

Comment:
The function should return `Option<&str>`.

To be more *rusty*, the whole match can be replaced with a [deref](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_deref), e.g.:
```rust
resource.message.as_deref()
```

Reply:
done